### PR TITLE
🎨 Set type as default sorting

### PR DESCRIPTION
**Pull request by AuroraHuang22**
網頁擷取_14-4-2023_152053_localhost

### CODE REVIEW

File: src/mixins/portfolio.js [hunk 1]
```diff
@@ -305,6 +305,27 @@ export const createPortfolioMixin = ({
...
+      if (
+        currentClassList.some(({ classId }) =>
+          this.nftClassMapOfNFTBook.has(classId)
+        ) ||
+        currentClassList.some(({ classId }) =>
+          this.nftClassMapOfOther.has(classId)
+        )
+      ) {
+        this.nftClassListOfCollectedSorting = NFT_CLASS_LIST_SORTING.TYPE;
+        this.nftClassListOfCreatedSorting = NFT_CLASS_LIST_SORTING.TYPE;
```
Comment (williamchong) [inline src/mixins/portfolio.js:322]: We should never have sideeffect in `computed` properties, they should be pure functional
i.e. no `this.xxx = yyy` inside any `computed` functions.

File: src/store/modules/nft.js [hunk 2]
```diff
@@ -229,10 +228,10 @@ const getters = {
         );
         if (isFeaturedCompareResult !== 0) return isFeaturedCompareResult;
       }
-      const isWritingNFTCompareResult = compareIsWritingNFT(getters, a, b);
-      if (isWritingNFTCompareResult !== 0) return isWritingNFTCompareResult;
       let X;
       let Y;
+      let XMetaData;
+      let YMetaData;
```
Comment (nwingt) [inline src/store/modules/nft.js:234]: Why using `let`? seems no reassignment

File: src/store/modules/nft.js [hunk 3]
```diff
@@ -181,16 +173,23 @@ const getters = {
...
```
Comment (nwingt) [inline src/store/modules/nft.js:179]: Why using `let`? seems no reassignment

File: src/mixins/portfolio.js [hunk 4]
```diff
@@ -302,9 +302,32 @@ export const createPortfolioMixin = ({
...
       }
     },
+    shouldCurrentSortingShowType() {
+      const currentClassList =
+        this.currentTab === tabOptions.collected
+          ? this.nftClassListOfCollectedInOrder
+          : this.nftClassListOfCreatedInOrder;
+
+      return (
+        (this.nftTypeFilter === 'ALL' &&
```
Comment (nwingt) [inline src/mixins/portfolio.js:312]:
```suggestion
        (this.nftTypeFilter === NFT_TYPE_FILTER_OPTIONS.ALL &&
```

**Review by nwingt — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**